src/components/Purchase/CheckoutComplete.js

```diff
@@ -0,0 +1,146 @@
+import axios from 'axios';
+import { useAuth } from '@akosasante/react-auth-context';
+
+const stripePromise = loadStripe(
+  'pk_test_51Nfj2pKnb0YvaiB3W0z31bbB7OPEnqem6wnpSbohudeiDkj1NUde9M5kgUwozzrfBqGqRpWU5ivDIInzWdt4q5zw00vMz7fc1c'
+);
+
+const CheckoutComplete = () => {
+  //---This is a temporarily data for testing purpose---
+  let n = 1;
+  //----------------------------------------------------
+  const location = useLocation();
+
+  const [clientSecret, setClientSecret] = useState(null);
+  const [isPaymentSuccessful, setIsPaymentSuccessful] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [orderData, setOrderData] = useState(null);
+
+  const { user } = useAuth();
+  const userEmail = user.user.email;
+
+  // Callback function to update isPaymentSuccessful
+  const updatePaymentStatus = (isSuccess) => {
+    setIsPaymentSuccessful(isSuccess);
+  };
+
+  useEffect(() => {
+    const queryParams = new URLSearchParams(location.search);
+    const clientSecret = queryParams.get('payment_intent_client_secret');
+    setClientSecret(clientSecret);
+
+    const orderDetails = async (id) => {
+      setIsLoading(true);
+      setErrorMessage('');
+      try {
+        const response = await axios.get(
+          `http://localhost:8000/api/v1/orders/${id}`,
+          {
+            withCredentials: true,
+            headers: {
+              'Content-Type': 'application/json',
+            },
+          }
+        );
+        setIsLoading(false);
+        setOrderData(response.data);
+        console.log(JSON.stringify(response.data));
+      } catch (error) {
+        setIsLoading(false);
+        setErrorMessage(
+          `An error occurred during loading order details with id ${id}`
+        );
+      }
+    };
+    orderDetails('64e950728af0bef9f0ced812');
+  }, [location]);
+
+  return (
+    <>
+      <Container className={style.mainContainer}>
+        <Grid container spacing={3}>
+          <Grid item xs={12} sm={6}>
+            <Paper className={`${style.paper} ${style.leftPaper}`}>
+              <Box mt="1rem">
+                <Typography variant="h4">
+                  <Elements stripe={stripePromise} options={{ clientSecret }}>
+                    <PaymentStatus
+                      clientSecret={clientSecret}
+                      updateStatus={updatePaymentStatus}
+                    />
+                  </Elements>
+                </Typography>
+              </Box>
+              {isPaymentSuccessful ? (
+                <>
+                  {isLoading ? (
+                    <Loader className="small-spinner" />
+                  ) : errorMessage !== '' ? (
+                    <>
+                      <Box mt="2rem">
+                        <Typography variant="h8">
+                          {/* {`Your order of ${orderData.order.orderItems.quantity} albums has been completed.`} */}
```

Comment:
The `orderItems` property will be a list of items: `[ {album: <album id>, quantity: <integer>, user: <user id> } ]`. The number of distinct albums ordered will be length of that list. Or if you want to indicate how many total items were purchased you may need to iterate through the list and add up all the quantities

Reply:
Done.